src/main/java/uk/gov/justice/hmpps/prison/service/AgencyService.java

```diff
@@ -303,7 +303,7 @@ private PrisonContactDetail getPrisonContactDetail(final uk.gov.justice.hmpps.pr
                 .build();
         } else {
             final var primaryAddress = prison.getAddresses().stream()
-                .filter(a -> "Y".equals(a.getPrimaryFlag())).findFirst().orElse(prison.getAddresses().get(0));
+                .filter(a -> "Y".equals(a.getPrimaryFlag())).findFirst().orElse(prison.getAddresses().iterator().next());
```

Comment:
really `get(0)` meant in this context "get any" so just use `next` on the set since we don't care which one we get